dropshot/src/schema_util.rs

```diff
@@ -814,6 +923,60 @@ mod test {
         );
     }
 
+    /// These cases are errors, but we want to provide good error messages for
+    /// them.
+    #[test]
+    fn test_schema2struct_with_enum_variants() {
+        #![allow(dead_code)]
+
+        #[derive(JsonSchema)]
+        enum People {
+            Alice,
+            Bob,
+            Mallory,
+        }
+
+        #[derive(JsonSchema)]
+        enum PeopleWithComments {
+            /// Alice
+            Alice,
+
+            /// Bob
+            Bob,
+
+            // Mallory doesn't get a doc comment but still gets reflected as a
+            // complex variant, because the other two have comments.
+            Mallory,
+        }
```

Comment:
The shape of the derived schemas are going to change depending on serde attributes such as `#[serde(tag = "tag")]` or `#[serde(tag = "tag", content = "content")]`

Reply:
I'll add tests to cover the other cases.

Reply:
Done.